File: app/schemas/interview.py
import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class InterviewCreate(BaseModel):
    application_id: uuid.UUID
    round_number: int
    scheduled_at: datetime | None = None
    conducted_by: uuid.UUID | None = None


class OutcomeUpdate(BaseModel):
    outcome: Literal["pending", "pass", "fail"]
    notes: str | None = None


class InterviewOut(BaseModel):
    id: uuid.UUID
    application_id: uuid.UUID
    round_number: int
    scheduled_at: datetime | None
    conducted_by: uuid.UUID | None
    outcome: str
    notes: str | None
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/interview.ts
export type InterviewOutcome = 'pending' | 'pass' | 'fail' | 'cancelled'

export interface InterviewCreate {
  application_id: string
  round_number: number
  scheduled_at?: string | null
  conducted_by?: string | null
}

export interface OutcomeUpdate {
  outcome: 'pending' | 'pass' | 'fail'
  notes?: string | null
}

export interface InterviewOut {
  id: string
  application_id: string
  round_number: number
  scheduled_at: string | null
  conducted_by: string | null
  outcome: InterviewOutcome
  notes: string | null
  created_at: string
}
